modules/azure/core/outputs.tf

```diff
@@ -0,0 +1,22 @@
+output "virtual_network" {
+  description = "Output for Azure Virtual Network"
+  value       = azurerm_virtual_network.this
+}
+
+output "subnets" {
+  description = "Output for Azure Virtual Network Subnets"
+  value = {
+    this = azurerm_subnet.this
```

Comment:
I dont know if "this" is the best name for the value. It would be a bit confusing what "this" compared to aks is.